test/linear_solver_routines/block_solver_helpers/proc_gentlemans_house.jl

```diff
@@ -0,0 +1,49 @@
+# This file is part of RLinearAlgebra.jl
+
+module ProceduralTestGent
+
+using Test, RLinearAlgebra, LinearAlgebra, Random
+
+@testset "Gent Projection -- Procedural" begin
+    A = rand(10,5)
+    x = rand(5)
+    b = A * x
+    G = RLinearAlgebra.GentData(A, 3)
+    # Check sizes of block matrix B 
+    m, n = size(G.B)
+    @test m == 3 + 1 + 5
+    @test n == 5 + 1
+
+    #Test block copy function 
+    RLinearAlgebra.copy_block_from_mat!(G.B, A, b, 1:3)
```

Comment:
Add comment that you are copying over rows 1 to 3. What happens if index range does not correspond to buffer size?

Reply:
What if the indices provided exceed the buffer size?

Reply:
I have added a test for the less than and greater than cases